src/main/javascript/src/pages/application/view/view.js

```diff
@@ -54,48 +54,79 @@ export class ApplicationView {
 	}
 
 	setApplicationStatus(id, status){
-		let allApplications = this.requested.concat(this.matchingInterests);
-		allApplications.forEach((application, index) => {
-			if(application.id === id){
-				application.status = status;
-			}
-			this.applicationManager.updateApplication(application);
-		});
+		if(this.isProf){
+			let allApplications = this.requested.concat(this.matchingInterests);
+			allApplications.forEach((application, index) => {
+				if(application.id === id){
+					application.status = status;
+				}
+				this.applicationManager.updateApplication(application);
+			});
+		} else if(this.isAdmin){
+			let allApplications = this.requested;
+			allApplications.forEach((application, index) => {
+				if(application.id === id){
+					let pos = this.requested.indexOf(application);
+					application.status = status;
+					if(!(this.statusFilter === "all")){
+						this.requested.splice(pos, 1);
+					}
+				}
+				this.applicationManager.updateApplication(application);
```

Comment:
I know I originally approved Luke's code that's like this, but man is this hard to read. Are you sure you can't use a filter() or a map() or reduce() or something?

Reply:
I'm not fluent enough in Javascript to know a better way to do it

Reply:
You can look at profile.js for inspiration : )
Further reading: https://overflowjs.com/posts/Map-Reduce-Filter-In-Javascript.html
What we're writing in is called ES6 javascript. (Ecmascript) We could also configure aurelia to compile our code written in JSX (typical react style) or typescript (typical angular style).

If this works, that's okay - no change required. Feel free to resolve this any time.

Reply:
This works as is so I'm not gonna change it so close to the deadline, but i've learned more about JS today